Adds uncompress module

Jmainguy opened the pull request:
This module allows for uncompressing a single compressed file
using either gzip, bzip2, or xz. I decided not to include zip
or 7zip as they always are archived as well,
and are better suited for the unarchive module.

robynbergeron commented:
@Jmainguy  -- Thanks for submitting this new module to Ansible Extras! This module is now in community review, a process that is open to all Ansible users. In order for this module to be approved, it must gain the following votes, which can be contributed by anyone other than the author of the new module:

“works_for_me”: If you have tested the module thoroughly, including testing of all of the module’s options, and if the module works for you, please add “works_for_me” in the comments.

“passes_guidelines”: If you have gone through the module guidelines and the module meets all of the requirements, please add “passes_guidelines” in the comments. Guidelines are available here: http://docs.ansible.com/developing_modules.html#module-checklist

“needs_revision”: If the module fails to work for you, or if it doesn’t meet guidelines, please add “needs_revision” in the comments with details about what needs to be fixed.

When a module has both “works_for_me” and “passes_guidelines” tags, we will promote the module for inclusion in Ansible Extras.  At this point, you will be expected to maintain the module by fixing bugs and evaluating pull requests in a timely manner. 

More information about the review process can be seen here: 
https://github.com/ansible/ansible-modules-extras/blob/devel/REVIEWERS.md

Thanks again for submitting your Ansible module!

mscherer commented on files/uncompress.py:
```diff
@@ -0,0 +1,265 @@
+    Also, xz does not support a destination flag, so we must use > to redirect stdout to file. Really lame.
+    """
+    cmd_path = module.get_bin_path('xz')
+    cmd = '%s -c -d %s > %s' % (cmd_path, src, dest)
```
Shouldn't we escape the path for shell metacharacters ? This seems rather unsafe, and while the stuff from a playbook is likely under a sysadmin control, we can't be sure someone will not take data coming from a non admin controlled location.

mscherer commented on files/uncompress.py:
```diff
@@ -0,0 +1,265 @@
+    """
+    cmd_path = module.get_bin_path('file')
+    cmd = "%s -b -i %s" % (cmd_path, src)
+    ftype = module.run_command(cmd, use_unsafe_shell=True)
```
Why do we need use_unsafe_shell here ?

mscherer commented on files/uncompress.py:
```diff
@@ -0,0 +1,265 @@
+                    f.write(data)
+                f.close()
+                src = package
+            except Exception, e:
```
There is no cleanup of the file if the download fail partially. IE, let's say that I start to download a 3G file and it break after 2G, I just have 2G of file in a directory which will not be cleaned by default (since that's ~/.ansible/tmp ).

mscherer replied:
So discussing on IRC, it should be cleaned automatically. However, looking at my own server, i see lots of left over in ~/.ansible/tmp, so I am not sure how everything is cleaned and if everything work fine for that.

mscherer commented on files/uncompress.py:
```diff
@@ -0,0 +1,265 @@
+    copy        = module.params['copy']
+    deep_check  = module.params['deep_check']
+    file_args   = module.load_file_common_arguments(module.params)
+    tempdir     = os.path.dirname(__file__)
```
Do we really want to make all download in the user home dir ? This might cause issue if people use it on NFS or have quota ?

But if we switch to /tmp, it should go in a specific tmp directory, and sometime /tmp can have quota too.

pilou- commented on files/uncompress.py:
```diff
@@ -0,0 +1,265 @@
+    """
+    try:
+        f_out = open(dest, 'wb')
+        f_in = gzip.open(src, 'rb')
```
`f_out.close()` is not called when `gzip.open` raises an exception. Same problem in the method below (`unbzip`).

pilou- commented on files/uncompress.py:
```diff
@@ -0,0 +1,265 @@
+    changed = False
+    if os.path.isfile(dest):
+        # This takes a long time
+        if deep_check is True:
```
use `if deep_check:` instead

Jmainguy commented:
@pilou- @mscherer Thank you for the comments. I have fixup up the file as recommended. No longer using > or unsafe_shell. Also added .close() in exceptions. The tmp_dir is set by the user, and deleted automatically unless they set it not to, so I am comfortable with the locations used, and lack of extra clean up

sivel commented:
Here are some additional notes found while performing an automated scan of this new module:

```
============================================================================
uncompress.py
============================================================================
ERROR: version_added should be 2.1. Currently 2.0
ERROR: No RETURN provided
```

Jmainguy commented:
@sivel What does no RETURN provided mean

bcoca commented:
document return values with a `RETURN='''` section, check stat module for example

Jmainguy commented:
@sivel @bcoca and in this module, I am only returning changed, so a RETUN = section is not needed correct?

Jmainguy commented:
Updated to reflect version 2.1

sivel commented:
I'm not positive.  Per recent discussions, all new modules must contain a `RETURN` although I am not sure.

Maybe the better question, is whether it would make sense to return more information from the module?

jctanner commented:
deprecated by http://docs.ansible.com/ansible/unarchive_module.html

jctanner commented:
Guess I misunderstood the context of this one. @Jmainguy if you still have an interest in this, please migrate it to ansible/ansible
